packages/x-date-pickers/src/internals/components/PickerFieldUI.tsx

```diff
@@ -78,8 +79,29 @@ export const cleanFieldResponse = <
     openPickerAriaLabel,
     textFieldProps: {
       ...other,
-      InputProps: { ...(InputProps ?? {}), readOnly },
-      inputProps: { ...(inputProps ?? {}), inputMode, onPaste, onKeyDown, ref: inputRef },
+      ...(materialMajor >= 6
+        ? {
+            slotProps: {
+              ...other?.slotProps,
+              input: {
+                ...other?.slotProps?.input,
```

Comment:
Could `other?.slotProps?.input` be a callback? 
If so we might have to use `resolveComponentProps`